### ISSUE TITLE
signInWithSocialWebUI throws a NullPointerException on 2.2.0 but not on 1.37.10

### ISSUE THREAD

**Issue by MrBenjoo** [aws-amplify/amplify-android#2282]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
[versions]
amplify = "2.2.0"

[libraries]
com.amplifyframework:core-kotlin:2.2.0
com.amplifyframework:aws-auth-cognito:2.2.0

[gradle plugin]
com.amplifyframework:amplify-tools-gradle-plugin:1.0.2
```

</details>

### Environment information

<details>

```
------------------------------------------------------------
Gradle 7.5
------------------------------------------------------------

Build time:   2022-07-14 12:48:15 UTC
Revision:     c7db7b958189ad2b0c1472b6fe663e6d654a5103

Kotlin:       1.6.21
Groovy:       3.0.10
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          1.8.0_275 (AdoptOpenJDK 25.275-b01)
OS:           Windows 10 10.0 amd64

```

</details>

### Please include any relevant guides or documentation you're referencing

https://docs.amplify.aws/lib/auth/social/q/platform/android/#update-androidmanifestxml

### Describe the bug

Calling ```Amplify.Auth.signInWithSocialWebUI(AuthProvider.google(), callingActivity)``` throws a nullpointerexception.

It works when using `core-kotlin:0.21.10` and `aws-auth-cognito:1.37.10`, i am able to sign in without problems. 

### Code Snippet

```kotlin
private suspend fun signInWithSocialWebUI(callingActivity: Activity) = try {
        Amplify.Auth.signInWithSocialWebUI(AuthProvider.google(), callingActivity)
        true
    } catch (exception: Exception) {
        false
    }
```

### Log output

<details>

java.lang.NullPointerException at com.amplifyframework.auth.cognito.actions.AuthorizationCognitoActions$initializeFetchAuthSession$$inlined$invoke$1.execute(Action.kt:89) at com.amplifyframework.statemachine.ConcurrentEffectExecutor$execute$1$1.invokeSuspend(ConcurrentEffectExecutor.kt:26) at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570) at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677) at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664) Suppressed: 
kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@c4f97ba, Dispatchers.Default]

</details>

## **amplifyconfiguration.json**
```
{
  "UserAgent": "aws-amplify-cli/2.0",
  "Version": "1.0",
  "auth": {
    "plugins": {
      "awsCognitoAuthPlugin": {
        "UserAgent": "aws-amplify-cli/0.1.0",
        "Version": "0.1.0",
        "IdentityManager": {
          "Default": {}
        },
        "CredentialsProvider": {
          "CognitoIdentity": {
            "Default": {
              "PoolId": "$poolIdCredentialsProvider",
              "Region": "$region"
            }
          }
        },
        "CognitoUserPool": {
          "Default": {
            "PoolId": "$poolIdCognito",
            "AppClientId": "$clientId",
            "AppClientSecret": "$clientSecret",
            "Region": "$region"
          }
        },
        "Auth": {
          "Default": {
            "OAuth": {
              "WebDomain": "$webDomain",
              "AppClientId": "$clientId",
              "AppClientSecret": "$clientSecret",
              "SignInRedirectURI": "$signInRedirectURI",
              "SignOutRedirectURI": "$signOutRedirectURI",
              "Scopes": [
                "email"
              ]
            },
            "authenticationFlowType": "USER_SRP_AUTH"
          }
        }
      }
    }
  }
}
```

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

### Additional information and screenshots

Also, I never got this exception on version 2.0.0.

**Comment by banji180** [comment]
@MrBenjoo we are not able to reproduce this issue. It seems in your case the token received on callback is null and that is not the expected behavior. We have a PR out to capture the NPE and display the error, but in the mean time, is other social plugins like facebook working?

**Comment by MrBenjoo** [comment]
I am getting the same exception after signing in with apple or facebook. Maybe it is unrelated, but i am seeing this in the logs each time the null exception is thrown:  

`win=Window{1bf7804 u0 com.android.project/com.amplifyframework.auth.cognito.activities.CustomTabsManagerActivity} destroySurfaces: appStopped=true win.mWindowRemovalAllowed=false win.mRemoveOnExit=false win.mViewVisibility=4 caller=com.android.server.wm.ActivityRecord.destroySurfaces:5867 com.android.server.wm.ActivityRecord.destroySurfaces:5848 com.android.server.wm.ActivityRecord.notifyAppStopped:5912 com.android.server.wm.ActivityRecord.activityStopped:6554 com.android.server.wm.ActivityClientController.activityStopped:254 android.app.IActivityClientController$Stub.onTransact:596 com.android.server.wm.ActivityClientController.onTransact:130`

**Comment by banji180** [comment]
ok thanks for the quick response, but I don't think this is the root cause. It seems the token is coming back as null, but since we are not able to reproduce, we are updating the SDK shortly to catch and display the **token is null** error. Once we release this update, we will update you.

**Comment by tjleing** [comment]
We have released library version [2.2.2](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.2.2) to attempt to resolve this issue.  Please let us know if that resolves your issue!

**Comment by MrBenjoo** [comment]
@tjleing Yes, it seems that it works now. Thanks!
